File: 1-two-sum/1-two-sum.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int,int> m;
        for(int i = 0;i<nums.size();i++)
        {
            m.insert({nums[i], i});
        }
        int left = 0;
        int right = nums.size()-1;
        for(int i = 0;i<nums.size();i++)
        {
            int find = target - nums[i];
            if(m.find(find)!=m.end() && m[find]!= i)
            {
                ans.push_back(i);
                ans.push_back(m[find]);
                break;
            }
        }
        return ans;
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> existingNums = new HashMap<>();
        int[] result = new int[2];
        
        for (int i = 0 ; i < nums.length ; i++) {
            
            if (existingNums.get(target - nums[i]) != null) {
                
                result[0] = existingNums.get(target - nums[i]);
                result[1] = i;
                break;
            } else {
                existingNums.put(nums[i], i);
            }
        }
        
        return result;
    }
}